token/js/test/e2e-2022/transferHook.test.ts

```diff
@@ -106,4 +133,96 @@ describe('transferHook', () => {
             expect(transferHook.authority).to.eql(PublicKey.default);
         }
     });
+    it('transferChecked', async () => {
+        const extraAccount = Keypair.generate().publicKey;
+        const keys: AccountMeta[] = [
+            { pubkey: pdaExtraAccountMeta, isSigner: false, isWritable: true },
+            { pubkey: mint, isSigner: false, isWritable: false },
+            { pubkey: payer.publicKey, isSigner: true, isWritable: false },
+            { pubkey: SystemProgram.programId, isSigner: false, isWritable: false },
+        ];
+
+        const data = Buffer.alloc(8 + 4 + ExtraAccountMetaLayout.span);
+        const discriminator = createHash('sha256')
+            .update('spl-transfer-hook-interface:initialize-extra-account-metas')
+            .digest()
+            .subarray(0, 8);
+        discriminator.copy(data);
+        ExtraAccountMetaListLayout.encode(
+            {
+                count: 1,
+                extraAccounts: [
+                    {
+                        discriminator: 0,
+                        addressConfig: extraAccount.toBuffer(),
+                        isSigner: false,
+                        isWritable: false,
+                    },
+                ],
+            },
+            data,
+            8
+        );
+
+        const initExtraAccountMetaInstruction = new TransactionInstruction({
+            keys,
+            data,
+            programId: TRANSFER_HOOK_TEST_PROGRAM_ID,
+        });
+
+        const setupTransaction = new Transaction().add(
+            initExtraAccountMetaInstruction,
+            createAssociatedTokenAccountInstruction(
+                payer.publicKey,
+                payerAta,
+                payer.publicKey,
+                mint,
+                TEST_PROGRAM_ID,
+                ASSOCIATED_TOKEN_PROGRAM_ID
+            ),
+            createMintToCheckedInstruction(
+                mint,
+                payerAta,
+                payer.publicKey,
+                5 * 10 ** TEST_TOKEN_DECIMALS,
+                TEST_TOKEN_DECIMALS,
+                [],
+                TEST_PROGRAM_ID
+            )
+        );
+
+        await sendAndConfirmTransaction(connection, setupTransaction, [payer]);
+
+        const transferCheckedInstruction = createTransferCheckedInstruction(
+            payerAta,
+            mint,
+            destinationAta,
+            payer.publicKey,
+            10 ** TEST_TOKEN_DECIMALS,
+            TEST_TOKEN_DECIMALS,
+            [],
+            TEST_PROGRAM_ID
+        );
+
+        const transferCheckedWithExtraAccountMeta = await addExtraAccountsToInstruction(
+            connection,
+            transferCheckedInstruction,
+            mint,
+            undefined,
+            TEST_PROGRAM_ID
+        );
```

Comment:
nit: what do you think about using the action helper that you created instead? that way we exercise a bit more code

Reply:
Yep makes sense!